src/main/java/org/ngafid/accounts/EmailType.java

```diff
@@ -167,13 +188,14 @@ public static void insertEmailTypesIntoDatabase(int userIDTarget) {
             removeOldEmailTypesFromDatabase(currentEmailTypes);
         }
 
-
         StringBuilder query = new StringBuilder();
         query.append("INSERT INTO email_preferences (user_id, email_type) ");
 
         //Generate "individual" email type queries
         for (EmailType emailType : EmailType.values()) {
-            query.append("SELECT ").append(userIDTarget).append(", '").append(emailType.getType()).append("' UNION ALL ");
+            String emailTypeKey = emailType.getType();
```

Comment:
Can probably replace this with `String::join`, which will join a list of strings with some delimiter. In this case `UNION ALL` will be your delimiter. This pattern comes up frequently enough that they made a utility function for it ;)